src/components/side_panel/split_to_columns_panel/split_to_columns_panel.ts

```diff
@@ -0,0 +1,136 @@
+import { Component, onMounted, onWillUpdateProps, useState } from "@odoo/owl";
+import { NEWLINE } from "../../../constants";
+import { interactiveSplitToColumns } from "../../../helpers/ui/split_to_columns_interactive";
+import { _lt } from "../../../translation";
+import { CommandResult, SpreadsheetChildEnv } from "../../../types/index";
+import { SplitToColumnsTerms } from "../../translations_terms";
+
+type SeparatorValue = "auto" | "custom" | " " | "," | ";" | typeof NEWLINE;
+
+interface Separator {
+  name: string;
+  value: SeparatorValue;
+}
+
+const SEPARATORS: Separator[] = [
+  { name: _lt("Detect automatically"), value: "auto" },
+  { name: _lt("Custom separator"), value: "custom" },
+  { name: _lt("Space"), value: " " },
+  { name: _lt("Comma"), value: "," },
+  { name: _lt("Semicolon"), value: ";" },
+  { name: _lt("Line Break"), value: NEWLINE },
+];
+
+interface Props {
+  onCloseSidePanel: () => void;
+}
+
+interface State {
+  separatorValue: SeparatorValue;
+  customSeparator: string;
+  addNewColumns: boolean;
+}
+
+export class SplitIntoColumnsPanel extends Component<Props, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-SplitIntoColumnsPanel";
+
+  state = useState<State>({ separatorValue: "auto", addNewColumns: false, customSeparator: "" });
+  errorMessages: string[] = [];
+  warningMessages: string[] = [];
+
+  setup() {
+    onWillUpdateProps(() => {
+      this.errorMessages = this.getErrorMessages();
```

Comment:
this won't be updated at a state change since onwillupdate will only be called by a parent.  the content of this callback should be extracted in another funcfion (like 'updateMessages') which could then be called by `updateAddNewColumnsCheckbox` and `updateCustomSeparator`, `onSeparatorChange`.

Reply:
Changed back to `get` methods, less error prone